### DISCUSSION TITLE
adjacency_matrix to accept #Nx2 arrays of edges

### DISCUSSION THREAD

**Discussion by maxschommer** [libigl/libigl#1879] [Ideas]
Right now, [adjacency_matrix](https://libigl.github.io/libigl-python-bindings/igl_docs/#adjacency_matrix) python bindings accept #Nx3 and #Nx4 arrays, representing triangle and tet meshes. I'd like to be able to build a sparse adjacency matrix using a list of edges as well, or an #Nx2 array, so I can get connected components from an edge array. It's clear this is easily supported in the C++ library, but just not in the bindings.

(this should probably be moved [here](libigl/libigl-python-bindings), but I can't move it)

**Comment by alecjacobson** [comment]
It should _just work_. So this a fix would be to update the doc in the header to write that it works for any simplex.

**Comment by maxschommer** [comment]
I made a PR over in the python bindings that seems to fix the issue. In the C++ header doc , it already seems to be correct and claims simplex generality. So it seems that the only issue was an overly stringent python shape check. This discussion should probably be closed:

https://github.com/libigl/libigl-python-bindings/pull/101
